### PR TITLE
docs: explain how to config and list of used insights

### PR DESCRIPTION
**Pull request by Jibbarth**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| Fixed tickets | #... 

TODO: 

- [x] Document all sniff/insight for code metrics
- [x] Document all sniff/insight for architecture metrics
- [x] Document all sniff/insight for complexity metrics
- [x] Document all sniff/insight for style metrics

### CODE REVIEW

**Comment by olivernybroe** [comment]
Nice initiative! 
When you add in the line length, could you add the parameters available for it also?
Related to #124.

File: docs/insights/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+::: tip
+To know the className of an Insights, launch `phpinsights` with `-v` option (verbose)
+:::
+
+## Configure Insights
+
+The `config` section allows you to refine default insight configuration.
+
+For example, to increase the line lenght limits:
```
Comment (olivernybroe) [inline docs/insights/README.md]: Great to use this as the example!

File: docs/insights/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+```php
+    'config' => [
+        \PHP_CodeSniffer\Standards\Generic\Sniffs\Files\LineLengthSniff::class => [
+            'lineLimit' => 120,
```
Comment (olivernybroe) [inline docs/insights/README.md]: When we add this in the docs section for the sniff, we need a nice way to explain the parameters.

Comment (Jibbarth) [reply docs/insights/README.md]: I added a badge *configurable*, and how it's configurable with default value in a <detail> html tag. 

Let me upload some captures :wink:

**Comment by Jibbarth** [comment]
Some screen capture :tada: 

FireShot Capture 002 - Configure Insights - PHP Insights - localhost
FireShot Capture 003 - Code - PHP Insights - localhost
FireShot Capture 004 - Code - PHP Insights - localhost

**Comment by olivernybroe** [comment]
If you could add just a quick comment on a config parameter, so other contributers later on knows how we are going to do it that would be awesome. When that is done, just go ahead and merge it in.

If you are up for it, then a quick tweet about the addition to the docs would be cool, else we can maybe get @nunomaduro to tweet it :)

**Comment by nunomaduro** [comment]
Thank you so much for your work in this pull request. 👌🏻

I am in a conference tour again this week - again. I will take a look at this soon as possible.

**Comment by Jibbarth** [comment]
> Could you add a example to one of the config parameters on how a comment on what the parameter means?

> If you could add just a quick comment on a config parameter

@olivernybroe I am really sorry, I don't understand what you want me to do :sweat: 

Can you indicate me directly on the files with the `suggestion` feature ?

File: docs/insights/style.md [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+
+**Insight Class**: `SlevomatCodingStandard\Sniffs\Namespaces\UnusedUsesSniff`
+
+<details>
+    <summary>Configuration</summary>
+
+```php
+\SlevomatCodingStandard\Sniffs\Namespaces\UnusedUsesSniff::class => [
+    'searchAnnotations' => false,
+    'ignoredAnnotationNames' => [], // case sensitive list of annotation names that the sniff should ignore (only the name is ignored, annotation content is still searched). Useful for name collisions like @testCase annotation and TestCase class.
```
Comment (olivernybroe) [inline docs/insights/style.md]: I meant exactly what you have done here actually. Where you added a comment on what the property meant :)

**Comment by Jibbarth** [comment]
@olivernybroe Thanks for the reviews :+1:

**Comment by olivernybroe** [comment]
@Jibbarth do you want to tweet about the new docs? I think it is something we should notify the users about as this is an awesome help for all the users.

**Comment by Jibbarth** [comment]
@olivernybroe It's done https://twitter.com/jibbarth/status/1138342628666331137

**Pull request merged**